connectivity/nfc/include/nfc/ndef/Record.h

```diff
@@ -165,17 +165,17 @@ struct Record {
      * containing it.
      */
     Record(
-        RecordType type,
```

Comment:
What's the issue here ? This is a well understood pattern to have constructor parameters named after the field they initialize. There's an issue opened for it in GCC: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78147 and clang address the problem by *not* raising a warning. You need an extra flag for that: `-Wshadow-field-in-constructor`. 

I'd be in favor of reverting this change.

Reply:
agreed!

Reply:
If there's a specific warning flag for "shadow field in constructor", so you can opt out for this one, but still warn about other nested scopes, then I largely agree.

I assumed this was being warned about because of a general shadowing warning, so decided to let it slide.

Reply:
`-Wshadow` doesn't trigger this warning with clang: https://godbolt.org/z/YYxrh7